modules/geo-layers/src/tile-layer/tile-2d-header.js

```diff
@@ -33,38 +38,36 @@ export default class Tile2DHeader {
     return result;
   }
 
-  loadData(getTileData) {
+  async _loadData(getTileData, requestScheduler) {
     const {x, y, z, bbox} = this;
-    if (!getTileData) {
-      return;
-    }
 
-    let tileData;
-    try {
-      tileData = getTileData({x, y, z, bbox});
-    } catch (err) {
-      this._isLoaded = true;
-      this.onTileError(err, this);
-      return;
-    }
+    const requestToken = await requestScheduler.scheduleRequest(this, tile => {
+      return tile.state === TILE_STATE_SELECTED ? 1 : -1;
+    });
 
-    if (!isPromise(tileData)) {
-      this.content = tileData;
-      this._isLoaded = true;
-      this.onTileLoad(this);
-      return;
-    }
-
-    this._loader = tileData
-      .then(buffers => {
-        this.content = buffers;
+    if (requestToken) {
+      this._isCancelled = false;
+      try {
+        const tileData = await getTileData({x, y, z, bbox});
+        requestToken.done();
+        this.content = tileData;
         this._isLoaded = true;
         this.onTileLoad(this);
-        return buffers;
-      })
-      .catch(err => {
+      } catch (err) {
+        requestToken.done();
```

Comment:
Is it a problem here that `requestToken.done()` could potentially be called twice?

I previously had `requestToken.done()` inside the `finally` block, but I thought it would be better to call it before `onTileLoad/onTileError` in case either of those were slow. But if an error were produced inside `onTileLoad`, then `requestToken.done` would be called twice.

Reply:
```js
let tileData;
let error;
try {
  tileData = await ...
} catch (err) {
  error = err;
} finally {
  requestToken.done();
  this._isLoaded = true;
}

if (error) {
 ...
}

Reply:
Apparently when you call `Promise.resolve(null)`, that propagates to the `catch` block with `err: null`. So I had to change that to:
```
} catch (err) {
  error = err || true;
} finally {
```

or else the `if (error)` block wouldn't run.